sonic_y_cable/y_cable.py

```diff
@@ -1237,3 +1243,37 @@ def get_nic_voltage_temp(physical_port):
         return -1
 
     return temp, voltage
+
+def firmware_upgrade(physical_port, fwfile):
+
+    """ This routine should facilitate complete firmware
+    upgrade of the Y cable on all the three ends of the
+    Y cable of the port specified.
+    All the components of the Y cable should be upgraded and committed
+    in their entirety by this single call subroutine.
+    This should return success code if firmware upgrade is successfull
+    and an error code otherwise.
+    Note that the error code on failure should reflect whether a rollback
+    or synchronization would be required or not required after calling this
+    routine.
+    Meaning for example if one of the ends of the Y cable could not
+    complete the upgrade and the cable could then require synchronization
+    after this subroutine call, then an error code stating that should
+    be the return value. Likewise if all the links are down
+    then that would not require a rollback of any sort and hence the error code 
+    should just reflect that.
+    Hence after calling this subroutine the caller can then determine
+    whether further action is necessary for firmware upgrade or not.
+
+    Args:
+        physical_port:
+             an Integer, the actual physical port connected to a Y cable
+        fwfile:
+             a File, the actual binary or executable which contains the firmware
```

Comment:
This should probably read `a String, a path to the binary file which contains the firmware image`

Reply:
fixed